app/scripts/views/components/dataTable/tableView.js

```diff
@@ -143,21 +156,110 @@
         autoColumnWidth: false,
         columnMaxWidth: 300,
         columnSorting: false,
+        sortBy: 'cases',
         selectedRows: selectedRows,
-        selectedGene: selectedGenes,
-        rowClickFunc: reactRowClickCallback,
-        geneClickFunc: reactGeneClickCallback,
-        selectButtonClickCallback: reactSubmitClickCallback,
+        rowClickFunc: function(data, selected) {
+          reactRowClickCallback(data, selected);
+          reactSubmitClickCallback();
+        },
         // sortBy: 'name',
         // sortDir: 'DESC',
         tableType: type_
       };
+      if (isMutatedGeneCna) {
+        _opts = _.extend(_opts, {
+          rowClickFunc: reactRowClickCallback,
+          selectedGene: selectedGenes,
+          geneClickFunc: reactGeneClickCallback,
+          selectButtonClickCallback: reactSubmitClickCallback
+        });
+      }
       var testElement = React.createElement(EnhancedFixedDataTableSpecial,
         _opts);
 
       ReactDOM.render(testElement, document.getElementById(chartId_));
     }
 
+    function initRegularTableData() {
+      var data = [];
+      _.each(categories_, function(category, name) {
+        for (var key in category) {
+          if (category.hasOwnProperty(key)) {
+            var datum = {
+              attr_id: key,
+              uniqueId: name,
+              attr_val: key === 'caseIds' ? category.caseIds.join(',') : category[key]
+            };
+            data.push(datum);
+          }
+        }
+      });
+      reactData_ = data;
+    }
+
+    function initCategories(_selectedMap) {
+      var colors = $.extend(true, [], iViz.util.getColors());
+      var hasSelectedCases = _.isObject(_selectedMap);
+      var numOfCases = 0;
+
+      if (hasSelectedCases) {
+        _.each(_selectedMap, function(datum) {
+          if (_.isArray(datum)) {
+            numOfCases += datum.length;
+          }
+        });
+      } else {
+        numOfCases = data_.length;
+      }
+
+      if (hasSelectedCases) {
```

Comment:
Any specific reason to have similar conditional statement and looping _selectedMap (lines 205,206 and 215,216)?